examples_tests/02.ComputeShader/main.cpp

```diff
@@ -27,11 +38,584 @@ void main()
+		device->createGPUPipelineLayout(nullptr, nullptr, core::smart_refctd_ptr(dsLayout));
+
+	core::smart_refctd_ptr<video::IGPUComputePipeline> pipeline =
+		device->createGPUComputePipeline(nullptr, core::smart_refctd_ptr(pipelineLayout),
+			core::smart_refctd_ptr(specializedShader));
+
+	// Todo(achal): Consider making it greater than swapchainImageCount
+	const uint32_t FRAMES_IN_FLIGHT = 2u;
+
+	core::smart_refctd_ptr<video::IGPUSemaphore> acquireSemaphores[FRAMES_IN_FLIGHT];
+	core::smart_refctd_ptr<video::IGPUSemaphore> releaseSemaphores[FRAMES_IN_FLIGHT];
+	core::smart_refctd_ptr<video::IGPUFence> frameFences[FRAMES_IN_FLIGHT];
+	for (uint32_t i = 0u; i < FRAMES_IN_FLIGHT; ++i)
+	{
+		acquireSemaphores[i] = device->createSemaphore();
+		releaseSemaphores[i] = device->createSemaphore();
+		frameFences[i] = device->createFence(video::IGPUFence::E_CREATE_FLAGS::ECF_SIGNALED_BIT);
+	}
+
+	// Record commands in commandBuffers here
+	for (uint32_t i = 0u; i < swapchainImageCount; ++i)
+	{
+		video::IGPUCommandBuffer::SImageMemoryBarrier undefToComputeTransitionBarrier;
+		undefToComputeTransitionBarrier.barrier.srcAccessMask = asset::EAF_TRANSFER_READ_BIT;
+		undefToComputeTransitionBarrier.barrier.dstAccessMask = asset::EAF_SHADER_READ_BIT;
+		undefToComputeTransitionBarrier.oldLayout = asset::EIL_UNDEFINED;
+		undefToComputeTransitionBarrier.newLayout = asset::EIL_GENERAL;
+		undefToComputeTransitionBarrier.srcQueueFamilyIndex = ~0u;
+		undefToComputeTransitionBarrier.dstQueueFamilyIndex = ~0u;
+		undefToComputeTransitionBarrier.image = *(swapchainImages.begin() + i);
+		undefToComputeTransitionBarrier.subresourceRange.aspectMask = asset::IImage::EAF_COLOR_BIT;
+		undefToComputeTransitionBarrier.subresourceRange.baseMipLevel = 0u;
+		undefToComputeTransitionBarrier.subresourceRange.levelCount = 1u;
+		undefToComputeTransitionBarrier.subresourceRange.baseArrayLayer = 0u;
+		undefToComputeTransitionBarrier.subresourceRange.layerCount = 1u;
+
+		video::IGPUCommandBuffer::SImageMemoryBarrier computeToPresentTransitionBarrier;
+		computeToPresentTransitionBarrier.barrier.srcAccessMask = asset::EAF_SHADER_WRITE_BIT;
+		computeToPresentTransitionBarrier.barrier.dstAccessMask = static_cast<asset::E_ACCESS_FLAGS>(0);
+		computeToPresentTransitionBarrier.oldLayout = asset::EIL_GENERAL;
+		computeToPresentTransitionBarrier.newLayout = asset::EIL_PRESENT_SRC_KHR;
+		computeToPresentTransitionBarrier.srcQueueFamilyIndex = ~0u;
+		computeToPresentTransitionBarrier.dstQueueFamilyIndex = ~0u;
+		computeToPresentTransitionBarrier.image = *(swapchainImages.begin() + i);
+		computeToPresentTransitionBarrier.subresourceRange.aspectMask = asset::IImage::EAF_COLOR_BIT;
+		computeToPresentTransitionBarrier.subresourceRange.baseMipLevel = 0u;
+		computeToPresentTransitionBarrier.subresourceRange.levelCount = 1u;
+		computeToPresentTransitionBarrier.subresourceRange.baseArrayLayer = 0u;
+		computeToPresentTransitionBarrier.subresourceRange.layerCount = 1u;
```

Comment:
you could factor this out to be common to both barriers

Reply:
resolved